### PR TITLE
fix(controller): add max wait time for load request

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description
close https://github.com/star-whale/starwhale/issues/2836
Add timeout for keylock.
BUG desc:The keyLock don't resolve the case of timeout, so it will be queued a long time even the client has been killed by the timeout error.

## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataReadManager.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ public Session generateSession(DataReadRequest request) {
     @Transactional
     public DataReadLog assignmentData(String consumerId, Session session) {
         var sid = session.getId();
-        var queue = sessionCache.computeIfAbsent(String.valueOf(sid), id -> new ConcurrentLinkedQueue<>());
+        var queue = sessionCache.computeIfAbsent(String.valueOf(sid), id -> new LinkedList<>());
 
         if (queue.isEmpty()) {
             queue.addAll(dataReadLogDao.selectTopsUnAssignedData(sid, cacheSize));
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataReadManager.java:124]: Release the key if empty?

Comment (goldenxinxing) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataReadManager.java:124]: done, use weakHashMap, auto released by GC

File: server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataReadManager.java [hunk 2]
```diff
@@ -41,17 +41,20 @@ public class DataReadManager {
...
                            DataIndexProvider dataIndexProvider,
-                           @Value("${sw.dataset.load.read.log-cache-size}") int cacheSize) {
+                           @Value("${sw.dataset.load.read.log-cache-size:1000}") int cacheSize,
+                           @Value("${sw.dataset.load.read.log-cache-timeout:24h}") String cacheTimeout
+    ) {
         this.sessionDao = sessionDao;
         this.dataReadLogDao = dataReadLogDao;
         this.dataIndexProvider = dataIndexProvider;
         this.cacheSize = cacheSize;
+        this.sessionCache = new LRUCache<>(16, DurationStyle.detectAndParse(cacheTimeout).toMillis());
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataReadManager.java:57]: Is 16 too small? make it configurable?

Comment (goldenxinxing) [reply server/controller/src/main/java/ai/starwhale/mlops/domain/dataset/dataloader/DataReadManager.java]: done

**Review by jialeicui — APPROVED** [review]
LGTM

**Pull request merged**